### PR TITLE
raftstore, metrics: add metrics for the number of peers in flashback state

### PR DESCRIPTION
**Pull request by JmPotato**
### What is changed and how it works?

Issue Number: ref #14748.

What's Changed:

```commit-message
Add metrics for the number of peers in the flashback state.
```

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

image

### Release note 

```release-note
None.
```

### CODE REVIEW

File: components/raftstore/src/store/fsm/apply.rs [hunk 1]
```diff
@@ -2987,8 +2987,16 @@ where
...
-        // Modify the region meta in memory.
         let mut region = self.region.clone();
+        // First time to enter or leave the flashback state, record the metrics.
+        if region.is_in_flashback != is_in_flashback {
+            if is_in_flashback {
+                PEER_IN_FLASHBACK_STATE.inc();
+            } else {
+                PEER_IN_FLASHBACK_STATE.dec();
+            }
+        }
```
Comment (overvenus) [inline components/raftstore/src/store/fsm/apply.rs:2998]: What if peer is merged or destroyed? Will they leak the metrics?

**Review by overvenus — COMMENTED** [review]
Could you add it to raftstore too?

File: components/raftstore/src/store/fsm/apply.rs (hunk 1 above)
Comment (JmPotato) [reply components/raftstore/src/store/fsm/apply.rs:2998]: Once a region is in the flashback state, there is no other raft command that could be proposed and applied, so it's ok to simply inc and dec the counter here.

File: components/raftstore-v2/src/operation/command/admin/flashback.rs [hunk 2]
```diff
@@ -7,7 +7,11 @@ use kvproto::{
     raft_serverpb::RegionLocalState,
 };
 use protobuf::Message;
-use raftstore::{coprocessor::RegionChangeReason, store::metrics::PEER_ADMIN_CMD_COUNTER, Result};
+use raftstore::{
```
Comment (BornChanger) [inline components/raftstore-v2/src/operation/command/admin/flashback.rs:10]: please also track the # of kv scanned to locate the keys to flashback and # of kv scanned to find the proper version.

File: components/raftstore/src/store/fsm/apply.rs (hunk 1 above)
Comment (overvenus) [reply components/raftstore/src/store/fsm/apply.rs]: LGTM, though there are some extreme cases that may leak the metrics, eg, a peer is isolated after preparing flashback, and rejoin cluster after it's been removed from raft group via confchange, in this case this peer may be destroyed without applying finish flashback.

**Review by overvenus — APPROVED** [review]

**Comment by JmPotato** [comment]
@Connor1996 PTAL, thx.

**Review by HuSharp — APPROVED** [review]
LGTM!

**Review by nolouch — APPROVED** [review]
lgtm

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**